### PR TITLE
syntax: Remove `Nt(Impl,Trait,Foreign)Item`

### PR DESCRIPTION
**Pull request by petrochenkov**
Follow-up to https://github.com/rust-lang/rust/pull/69366.
r? @Centril

### CODE REVIEW

**Review by Centril — COMMENTED** [review]
Thanks. I wonder whether we have a test for trait items specifically that exercises the interaction with `NtTraitItem` and parameter names on Rust 2015. It doesn't seem like this would cause any problems re. that as `Annotatable::TraitItem`s would still cause us to parse using `parse_trait_item`. But could you double check? r=me with that and these comments considered.

**Comment by petrochenkov** [comment]
>I wonder whether we have a test for trait items specifically that exercises the interaction with NtTraitItem and parameter names on Rust 2015.

These nonterminals shouldn't be observable during parsing, they are an implementation details of, well, not even expansion, but pretty-printing.
If tokens in a `TokenStream` are packed into a nonterminal with an AST piece inside, then they are printed more nicely using the AST pretty-printer.

After https://github.com/rust-lang/rust/pull/62667 tokens streams themselves are printed pretty nicely, so I want to try removing this piece of code in `expand.rs` entirely in a follow-up PR.

File: src/libsyntax/token.rs [hunk 1]
```diff
@@ -712,12 +712,6 @@ pub enum Nonterminal {
     NtPath(ast::Path),
     NtVis(ast::Visibility),
     NtTT(TokenTree),
-    // Used only for passing items to proc macro attributes (they are not
-    // strictly necessary for that, `Annotatable` can be converted into
-    // tokens directly, but doing that naively regresses pretty-printing).
```
[the remark this answers is not in the document]
Comment (petrochenkov) [reply src/libsyntax/token.rs]: `NtItem` is actually used.
I'll better move this to the corresponding code in `expand.rs`.

**Review by Centril — APPROVED** [review]

**Comment by petrochenkov** [comment]
Updated.

**Pull request merged**